### PR TITLE
build: introduce and use `swift_target_link_search_directories`

### PR DESCRIPTION
**Pull request by compnerd**
Introduce a swift_target_link_search_directories which allows you to
append a search path to the linker.  This interface also allows a tweak
to make the library search directory addition more portable across
compilers (e.g. cl vs clang).

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
CC @llvm-beanz @hughbe

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please smoke test OS X platform

**Comment by compnerd** [comment]
@swift-ci please test OS X platform

**Comment by hughbe** [comment]
Good change, thanks

File: cmake/modules/AddSwift.cmake [hunk 1]
```diff
@@ -294,7 +294,7 @@ endfunction()
 function(_add_variant_link_flags)
   set(oneValueArgs SDK ARCH BUILD_TYPE ENABLE_ASSERTIONS ANALYZE_CODE_COVERAGE
   DEPLOYMENT_VERSION_OSX DEPLOYMENT_VERSION_IOS DEPLOYMENT_VERSION_TVOS DEPLOYMENT_VERSION_WATCHOS
-  RESULT_VAR_NAME ENABLE_LTO LTO_OBJECT_NAME)
+  RESULT_VAR_NAME ENABLE_LTO LTO_OBJECT_NAME LIBRARY_SEARCH_DIRECTORIES)
```
Comment (llvm-beanz) [inline cmake/modules/AddSwift.cmake]: I think it would be better if this were a multi-value argument rather than a single value argument.

**Review by llvm-beanz — DISMISSED** [review]
One comment below, otherwise LGTM.

File: cmake/modules/AddSwift.cmake (hunk 1 above)
Comment (compnerd) [inline cmake/modules/AddSwift.cmake]: @llvm-beanz thats an out parameter, which is the variable that gets the list.  Im not sure what you are suggesting here.

Comment (llvm-beanz) [inline cmake/modules/AddSwift.cmake]: I misunderstood something here. It actually looks to me like you're using it as both an in and out parameter. That is confusing and really gross, but that is how the other options work, so I guess it is okay.

**Pull request merged**